test/microapps-ui-xcm/microapps-ui-main.check.pools.test.ts

```diff
@@ -0,0 +1,230 @@
+      getEnvironmentRequiredVars().mnemonicPolkadot,
+    );
+
+    testUser1.addAsset(KSM_ASSET_ID);
+    testUser1.addAsset(MGA_ASSET_ID);
+    await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+    await setupPage(driver);
+    await connectWallet(driver, "Polkadot", acc_name);
+  });
+
+  it("All promoted pools exist in app", async () => {
+    const promotedPoolsRewards =
+      await api.query.proofOfStake.promotedPoolRewards();
+    const promotedPools = JSON.parse(
+      JSON.stringify(Object.keys(promotedPoolsRewards.toHuman())),
+    );
+    const promotedPoolsLength = promotedPools.length;
+    const promotedPoolsInfo = [];
+    for (let i = 0; i < promotedPoolsLength; i++) {
+      const poolName = await sdk.query.getLiquidityPool(promotedPools[i]);
+      const firstTokenId = await sdk.query.getTokenInfo(poolName[0]);
+      const secondTokenId = await sdk.query.getTokenInfo(poolName[1]);
+      const poolData = {
+        poolID: promotedPools[i],
+        firstToken: firstTokenId.symbol,
+        secondToken: secondTokenId.symbol,
+      };
+      promotedPoolsInfo.push(poolData);
+    }
+
+    const sidebar = new Sidebar(driver);
+    await sidebar.clickNavLiqPools();
+    const poolsList = await new LiqPools(driver);
+    await comparePoolsLists(promotedPoolsInfo, poolsList);
+  });
+
+  it("All liquidity pools exist in app", async () => {
+    const liquidityAssets = await api.query.xyk.liquidityAssets.entries();
+    const liquidityAssetsLength = liquidityAssets.length;
+    const liquidityPoolsInfo = [];
+    for (let i = 0; i < liquidityAssetsLength; i++) {
+      const liquidityAsset = JSON.parse(
+        JSON.stringify(liquidityAssets[i][1].value),
+      );
+      const poolName = await sdk.query.getLiquidityPool(liquidityAsset);
+      if (poolName[1] !== "2") {
+        const firstTokenInfo = await sdk.query.getTokenInfo(poolName[0]);
+        const secondTokenInfo = await sdk.query.getTokenInfo(poolName[1]);
+        const poolData = {
+          poolID: liquidityAsset,
+          firstToken: firstTokenInfo.symbol,
+          secondToken: secondTokenInfo.symbol,
+        };
+        liquidityPoolsInfo.push(poolData);
+      }
+    }
+
+    const sidebar = new Sidebar(driver);
+    await sidebar.clickNavLiqPools();
+    const poolsList = await new LiqPools(driver);
+    await poolsList.clickAllPoolsTab();
+    await comparePoolsLists(liquidityPoolsInfo, poolsList);
+  });
+
+  afterEach(async () => {
+    const session = await driver.getSession();
+    await addExtraLogs(
+      driver,
+      expect.getState().currentTestName + " - " + session.getId(),
+    );
+  });
+
+  afterAll(async () => {
+    StashServiceMockSingleton.getInstance().stopServer();
+    await kusama.teardown();
+    await mangata.teardown();
+    const api = getApi();
+    await api.disconnect();
+    await driver.quit();
+    DriverBuilder.destroy();
+  });
+});
+
+async function comparePoolsLists(bePoolsInfo: any, poolsList: LiqPools) {
```

Comment:
split this function in two, one to obtain the fePoolInfo and the second to compare them.

Reply:
Done